CHANGES.md

```diff
@@ -2,6 +2,434 @@
+* optimized memory management through new jemalloc upgrades and features
+* optimized reductions by using an atomic spin lock rather than a sync var
+* enabled the bulk transfer of strided array slices by default
+* implemented 'sync' variables natively for 'qthreads' tasking
+* optimized functions returning references to avoid widening them when possible
+* optimized on-statements that end up being local
+* optimized the implementation of decrementing remote task counters
+* stopped heap-allocating variables due to on-clauses for 'fifo' and 'muxed'
+* fixed a potential performance issue when growing/shrinking arrays as vectors
+* made modest improvements to the performance of associative domains
+* optimized the implementation of .re and .im for complex values
+* optimized base**exp operations when 'base' is a param power of two
+
+Memory Improvements
+-------------------
+* obtained memory allocation improvements through 'jemalloc' upgrades and usage
+* closed memory leaks caused by 'sync'/'single' variables
+* closed a memory leak relating to 'args' arguments sent to main()
+* closed a memory leak caused by module-scope detuple variable declarations
+
+Documentation
+-------------
+* added the primer example codes to the online documentation
+  (see TODO)
+* added a new primer example for modules and 'use' statements
+  (see doc/release/examples/primers/modules.chpl)
+* added support for console-/man-based chpldocumentation
+  (see TODO)
+* added a number of new sections to the user's guide
+  (see http://chapel.cray.com/docs/1.14/users-guide/index.html)
+* significantly re-worked the documentation for multi-locale execution
+  (see TODO)
+* moved information about execution on UDP-based systems into its own document
+  (see http://chapel.cray.com/docs/master/platforms/udp.html)
+* added usage ('use ...') information to chpldoc-generated module documentation
+  (e.g., see http://chapel.cray.com/docs/1.14/modules/standard/Assert.html)
+* added documentation for the 'chplvis' file format
+  (see $CHPL_HOME/tools/chplvis/TextDataFormat.txt)
+* updated the steps required to build Chapel with newer HDFS sources
+  (see http://chapel.cray.com/docs/1.14/modules/packages/HDFS.html and
+   http://chapel.cray.com/docs/master/technotes/auxIO.html#setting-up-hdfs)
+* added an archive of past language specification versions to the online docs
+  (see http://chapel.cray.com/docs/1.14/language/archivedSpecs.html))
+* updated and fixed minor issues in the language specification
+* fixed a plethora of spelling mistakes throughout the documentation
+* fixed some online documentation formatting errors
+
+Example Codes
+-------------
+* added all remaining Computer Language Benchmark Game programs
+  (see $CHPL_HOME/examples/benchmarks/shootout/*)
+* significantly improved existing Computer Language Benchmark Game programs
+  (see $CHPL_HOME/examples/benchmarks/shootout/*)
+* made improvements to LCALS for style and performance
```

Comment:
Is it worth mentioning that all LCALS kernels and variants are performance tested and tracked now?